src/arena.c

```diff
@@ -1350,6 +1353,9 @@ arena_tcache_fill_small(tsdn_t *tsdn, arena_t *arena, tcache_t *tcache,
 		tbin->tstats.nrequests = 0;
 	}
 	malloc_mutex_unlock(tsdn, &bin->lock);
+	if (fresh_slab) {
```

Comment:
Let's use `if (dalloc)` directly, and get rid of the branch @ L1326?

Reply:
I don't think we can do this, since dalloc is not set after fresh_slab, we may be using slabcur again instead, yet we still need to free fresh_slab since it is unused.